src/ProductBundle/Controller/CollectionController.php

```diff
@@ -21,92 +21,57 @@
 class CollectionController extends Controller
 {
     /**
-     * List the main collections.
-     *
-     * @return Response
-     */
-    public function indexAction(Request $request)
-    {
-        $pager = $this->get('sonata.classification.manager.collection')
-            ->getRootCollectionsPager($request->get('page'));
-
-        return $this->render('@SonataProduct/Collection/index.html.twig', [
-            'pager' => $pager,
-        ]);
-    }
-
-    /**
-     * Display one collection.
+     * List the product related to one collection.
      *
      * @param $collectionId
-     * @param $slug
-     *
-     * @throws NotFoundHttpException
      *
      * @return Response
      */
-    public function viewAction($collectionId, $slug)
+    public function listProductsAction(Request $request, $collectionId = null)
```

Comment:
```suggestion
    public function index(Request $request)
```
It is simmilar to CatalogController:index, so make the same logic